prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirToHl7ConverterTests.kt

```diff
@@ -561,4 +562,54 @@ class FhirToHl7ConverterTests {
         val shouldBeTruncated = terser.get("/PATIENT_RESULT/PATIENT/PID-5-1")
         assertEquals(shouldBeTruncated.length, 194)
     }
+
+    @Test
+    fun `test truncation logic for OML_O21`() {
+        val mockBundle = mockk<Bundle>()
+        val mockSchema = mockk<ConverterSchema>()
+        val terser = Terser(OML_O21())
+
+        // dummy config with just truncation config set up
+        val hl7Config = UnitTestUtils.createConfig(
+            truncateHl7Fields = "OBX-18-1",
+            truncateHDNamespaceIds = true
+        )
+
+        val config = HL7TranslationConfig(
+            hl7Config,
+            receiver = null
+        )
+
+        val contextWithConfig = FhirToHl7Context(
+            CustomFhirPathFunctions(),
+            config,
+            CustomTranslationFunctions()
+        )
+
+        val customContext = CustomContext(
+            mockBundle,
+            mockBundle,
+            config = config,
+            translationFunctions = CustomTranslationFunctions()
+        )
+
+        val converter = FhirToHl7Converter(
+            mockSchema,
+            terser = terser,
+            context = contextWithConfig
+        )
+
+        // should truncate to 194
+        val value = "x".repeat(500)
+
+        val element = ConverterSchemaElement(
+            "name",
+            required = true,
+            hl7Spec = listOf("/ORDER/OBSERVATION_REQUEST/OBSERVATION/OBX-18-1")
+        )
+        converter.setHl7Value(element, value, customContext)
+
+        val shouldBeTruncated = terser.get("/ORDER/OBSERVATION_REQUEST/OBSERVATION/OBX-18-1")
+        assertEquals(shouldBeTruncated.length, 199)
```

Comment:
199? Comment says 194, though?

Reply:
whoops, I need to double check on some comments as much of the unit tests got copy/pasted with some values swapped

Reply:
199 is right I'll change the comment

![Screenshot 2023-09-20 at 1 26 21 PM](https://github.com/CDCgov/prime-reportstream/assets/1209453/e4c1eaa7-a7e2-467d-891d-87ebf3636872)